src/utils/flags.ts

```diff
@@ -33,11 +40,115 @@ const parsePathInfo = async (input: string, exists: boolean): Promise<PathInfo>
   return { type: 'file', path: input };
 };
 
-export const fileOrDirFlag = (
-  opts: { exists?: boolean } & Partial<Interfaces.OptionFlag<PathInfo | undefined>> = {}
-): Interfaces.OptionFlag<PathInfo | undefined> => {
+type EnsureFsFlagOptions = {
+  flagName: string;
+  path: string;
+  type: 'dir' | 'file' | 'any';
+  throwOnENOENT?: boolean;
+};
+
+interface FsError extends Error {
+  code: string;
+}
+
+/**
+ * Ensures command flags that are file system paths are set properly before
+ * continuing command execution.  Can also create directories that don't yet
+ * exist in the path.
+ *
+ * @param options defines the path to resolve and the expectations
+ * @returns the resolved flag path
+ */
+function ensureFlagPath(options: EnsureFsFlagOptions): string {
+  const { flagName, path, type, throwOnENOENT } = options;
+
+  const trimmedPath = path?.trim();
+  let resolvedPath: string;
+  if (trimmedPath?.length) {
+    resolvedPath = resolve(trimmedPath);
+  }
```

Comment:
1. avoids `let`
2. explicit that resolvedPath could be `undefined` (I'm not sure that's how it's supposed to work, but that's possible in both the existing and suggested code.

```suggestion
  const resolvedPath =  trimmedPath?.length ? resolve(trimmedPath) : undefined;
```